ARCHITECTURE.md

```diff
@@ -68,16 +68,19 @@ the server understands ("SolidDatasets"), and those that do not ("Files").
 The main difference between the two is that, by virtue of the server
 understanding the structure of the former, we can do partial updates (i.e.
 PATCH requests) and retrieve the data in different formats (currently, the
-spec mandates that servers can serve such data as at least Turtle and JSON-LD).
+spec mandates that servers can receive/return such data as at least Turtle and JSON-LD).
 
-Thus, data stored as Turtle or JSON-LD can be fetched as a SolidDataset; the
+Thus, data requested as Turtle or JSON-LD can be fetched as a SolidDataset; the
 rest are just treated as regular Files: while it might be possible to parse and
 manipulate them using other libraries, solid-client just allows downloads and
 uploads, and not more specific operations like reading values for a given
-property. This applies to binary file types like JPEG, WebM and `.txt`, but also
-to files containing structured data in a non-RDF format, like JSON, XML or
-OpenDocument, or even to files containing structured data in an RDF format not
-understood by the server, such as RDFa or RDF-XML.
+property. This applies to binary file types like JPEG, WebM and `.txt`.
+When it comes to files containing structured data in a non-RDF format (e.g.: JSON, 
+XML or OpenDocument) or files containing structured data in an RDF format not
+understood by the server (e.g.: RDFa or RDF-XML) is up to the server to accept 
+a payload of RDF/XML and treat it as a binary resource (as is being alluded to here), 
```

Comment:
When I said "as alluded to here" I meant the original text was alluding to server's always storing that payload as binary - this suggested change isn't alluding that anymore :) !
```suggestion
that payload of RDF/XML and treat it as a binary resource, 
```